common/src/main/java/com/wynntils/core/statistics/StatisticsManager.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.statistics;
+
+import com.wynntils.core.components.Manager;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.storage.Storage;
+import com.wynntils.models.character.event.CharacterUpdateEvent;
+import com.wynntils.models.damage.type.DamageDealtEvent;
+import com.wynntils.models.spells.event.SpellEvent;
+import com.wynntils.models.stats.type.DamageType;
+import com.wynntils.models.worlds.event.WorldStateEvent;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+// This should really be an "ExternalModel"...
+public final class StatisticsManager extends Manager {
```

Comment:
The other "this really should be an external model" classes are all modes, what's different here?

Reply:
So, my initial idea was that we should sprinkle `Managers.Statistics.increaseStatistic()` all around the code base, and I did not want models to be dependent on another model, but a manager would be fine.

But then when I implemented the two initial statistics, it turned out that they were just listening to events. And maybe we should require that this is the only way statistics is allowed to be used, that if we want to have statistics, we need to send an event, which can then be listened to by the statistics manager/model.

I am worried that this will raise the bar too much and make it cumbersome to add new statistics, instead of just a single line of code. But otoh, it might also increase code quality, since there might be other potential listeners for that event. And having a "listener only" interface into the statistics would make it more self-sustaining, and allow it to be a model.

I could go either way here, so I'm interested in your opinion.